json-rpc/eth_call.md

```diff
@@ -0,0 +1,113 @@
+
+
+# `eth_call`
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Specification | Description 
+--|--
+1         | `eth_call` MUST NOT affect on-chain state|
+2         | `eth_call` MUST NOT accept requests that are not in valid JSON format
+3         | `eth_call` MUST return a response in valid JSON format
+4         | `eth_call` MUST accept a `params` parameter [JSON Array](https://tools.ietf.org/html/rfc7159#section-5) of length 2, and its first item containing a [JSON Object](https://tools.ietf.org/html/rfc7159#section-4) with keys known here as the `input parameters`
+4.1       | `eth_call` MUST accept an input parameter `from` of type `DATA` [!]
+4.1.1     | `eth_call` MUST accept requests with the from input parameter undefined
+4.1.1.1   | `eth_call` MUST consider CALLER account to be `0x0000000000000000000000000000000000000000000000000000000000000000` if `from` is not defined
+4.1.2     | `eth_call` MUST accept requests with the `from` input parameter defined
+4.1.2.1   | `eth_call` MUST only accept requests with the `from` parameter defined as a hex string with length 40
+4.1.2.1.1 | `eth_call` SHOULD throw error `-32602` [!] on improper length with message as `invalid argument 0: hex string has length {length}, want 40 for common.Address`
+4.1.2.2   | `eth_call` MUST NOT accept `from` hex strings with invalid characters
```

Comment:
As mentioned above, I think parsing errors like this should be generalized at a higher layer, otherwise we'll have to re-specify them over and over again.  We should define an `ADDRESS` type for the JSON-RPC protocol and define how parsing errors should be handled in a top level spec, then in the individual specs like this one we can just say "`from` **MUST** be of type `ADDRESS` and not have to specify all of the details of what that means.